nvtabular/io/fsspec_utils.py

```diff
@@ -0,0 +1,331 @@
+    buf = np.zeros(file_size, dtype="b")
+    if byte_ranges:
+
+        # Optimize/merge the ranges
+        byte_ranges = _merge_ranges(
+            byte_ranges,
+            max_block=bytes_per_thread,
+            max_gap=max_gap,
+        )
+
+        # Call multi-threaded data transfer of
+        # remote byte-ranges to local buffer
+        _read_byte_ranges(
+            path_or_fob,
+            byte_ranges,
+            buf,
+            fs,
+            **kwargs,
+        )
+
+        # Add Header & Footer bytes
+        if footer is not None:
+            footer_size = len(footer)
+            buf[-footer_size:] = np.frombuffer(footer[-footer_size:], dtype="b")
+
+        # Add parquet magic bytes (optional)
+        if add_par1_magic:
+            buf[:4] = np.frombuffer(b"PAR1", dtype="b")
+            if footer is None:
+                buf[-4:] = np.frombuffer(b"PAR1", dtype="b")
+
+    else:
+        byte_ranges = [
+            (b, min(bytes_per_thread, file_size - b)) for b in range(0, file_size, bytes_per_thread)
+        ]
+        _read_byte_ranges(
+            path_or_fob,
+            byte_ranges,
+            buf,
+            fs,
+            **kwargs,
+        )
+
+    return buf.tobytes()
+
+
+def _merge_ranges(byte_ranges, max_block=256_000_000, max_gap=64_000):
+    # Simple utility to merge small/adjacent byte ranges
+    new_ranges = []
+    if not byte_ranges:
+        # Early return
+        return new_ranges
+
+    offset, size = byte_ranges[0]
+    for (new_offset, new_size) in byte_ranges[1:]:
+        gap = new_offset - (offset + size)
+        if gap > max_gap or (size + new_size + gap) > max_block:
+            # Gap is too large or total read is too large
+            new_ranges.append((offset, size))
+            offset = new_offset
+            size = new_size
+            continue
+        size += new_size + gap
+    new_ranges.append((offset, size))
+    return new_ranges
+
+
+def _assign_block(fs, path_or_fob, local_buffer, offset, nbytes):
+    with fs.open(path_or_fob, mode="rb", cache_type="none") as fob:
+        fob.seek(offset)
+        local_buffer[offset : offset + nbytes] = np.frombuffer(
+            fob.read(nbytes),
+            dtype="b",
+        )
+
+
+def _read_byte_ranges(
+    path_or_fob,
+    ranges,
+    local_buffer,
+    fs,
+    **kwargs,
+):
+
+    # If we have a fs object that supports `cat_ranges`,
+    # the ranges should be collected concurrently
+    if hasattr(fs, "cat_ranges"):
+        paths, starts, ends = [], [], []
+        for offset, nbytes in ranges:
+            paths.append(path_or_fob)
+            starts.append(offset)
+            ends.append(offset + nbytes)
+        blocks = fs.cat_ranges(paths, starts, ends)
+        for block, (offset, nbytes) in zip(blocks, ranges):
+            local_buffer[offset : offset + nbytes] = np.frombuffer(
+                block,
+                dtype="b",
+            )
+        return
```

Comment:
```suggestion
```
Lets remove this. special logic util fsspec is actually released with [fsspec#744](https://github.com/intake/filesystem_spec/pull/744) in place.